### PR TITLE
Streamline the usage of the ResourcesReconciler

### PR DESCRIPTION
**Pull request by slinkydeveloper**
## Proposed Changes

- :broom: Moved inner class `ResourcesReconcilerImpl.Builder` to `ResourcesReconcilerBuilder`
- :broom: Moved `ResourcesReconcilerImpl.builder()` to `ResourcesReconciler.builder()`
- :gift: Now `ResourcesReconcilerBuilder` provides a `buildAndListen` method which builds directly the `ResourcesReconcilerMessageHandler` as well

### CODE REVIEW

File: data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/ConsumerDeployerVerticle.java [hunk 1]
```diff
@@ -67,13 +62,10 @@ public ConsumerDeployerVerticle(
...
-      ResourcesReconcilerImpl
-        .builder()
-        .watchEgress(this)
-        .build()
-    );
+    this.messageConsumer = ResourcesReconciler
+      .builder()
+      .watchEgress(this)
+      .buildAndListen(vertx);
   }
```
Comment (matzew) [inline data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/ConsumerDeployerVerticle.java:69]: looks better now.

**Review by matzew — APPROVED** [review]
/lgtm
/approve

**Comment by slinkydeveloper** [comment]
/unhold

My bad, I forgot to fix some tests using the old builder method. Now it should be fine

**Review by pierDipi — APPROVED** [review]

**Pull request merged**